src/pbt/__init__.py

```diff
@@ -30,104 +43,233 @@ def __init__(self, path_root: str):
 
             path_job_absolute = os.path.join(os.path.join(self.path_root, path_job), 'code')
             path_job_definition = os.path.join(path_job_absolute, 'databricks-job.json')
 
-            job_definition = {}
             with open(path_job_definition, 'r') as _in:
                 job_definition = json.load(_in)
 
             components = job_definition['components']
+
             for component in components:
                 if 'PipelineComponent' in component:
                     pipeline_uri = component['PipelineComponent']['id']
 
-                    uri_pattern = '([0-9]+)/([-_.A-Za-z0-9 /]+)'
+                    # Matches project_id/pipelines/pipeline_name or pipelines/pipeline_name
+                    # group(2) should return pipelines/pipeline_name
+                    uri_pattern = '([0-9]*)(pipelines/[-_.A-Za-z0-9 /]+)'
                     pipeline_id = re.search(uri_pattern, pipeline_uri).group(2)
 
                     source_path = self.pipelines_build_path[pipeline_id]['source_absolute']
                     target_path = component['PipelineComponent']['path']
 
                     print('    Uploading %s to %s' % (self.pipelines_build_path[pipeline_id]['source'], target_path))
-                    self.dbfs_service.put(target_path, overwrite=True, src_path=source_path)
 
-                    self.pipelines_build_path[pipeline_id]['uploaded'] = True
+                    try:
+                        self.dbfs_service.put(target_path, overwrite=True, src_path=source_path)
+                        self.pipelines_build_path[pipeline_id]['uploaded'] = True
+                    except Exception as e:
+                        self.pipelines_build_path[pipeline_id]['uploaded'] = False
+                        pipelines_upload_failures[pipeline_id].append(str(e))
+
+            if len(pipelines_upload_failures) > 0:
+                print("\n[bold red]❌ Upload failed for one or more pipelines [/bold red]")
+                for pipeline_id, errors in pipelines_upload_failures.items():
+                    print('\n[bold red]❌ %s: Exceptions: %s [/bold red]' % (pipeline_id, "\n".join(errors)))
+
+                # Process next job
+                continue
 
             job_request = job_definition['request']['CreateNewJobRequest']
             job_request['version'] = '2.1'
 
             limit = 25
             current_offset = 0
             found_job = None
-            while found_job is None:
-                response = self.jobs_service.list_jobs(limit=limit, offset=current_offset, version='2.1')
-                current_offset += limit
-
-                found_jobs = response['jobs'] if 'jobs' in response else []
-                for potential_found_job in found_jobs:
-                    if potential_found_job['settings']['name'] == job_request['name']:
-                        found_job = potential_found_job
+            try:
+                while found_job is None:
```

Comment:
Do we want to consider having a retryLimit or maxTimeout for querying here?

Reply:
Will need to write my own wrapper for the retry and timeout as existing sdk doesnt have it, will see later if I need it. Added a log line for now.